e2e/e2e.go

```diff
@@ -0,0 +1,241 @@
+		}
+
+		f, err := fs.Open("service.yaml")
+		if err != nil {
+			deps[name] = true
+			log.Warn().Err(err).Msg("Failed to find service.yaml, will not try to calculate dependencies of this service")
+			return nil
+		}
+
+		var s *Service
+		err = yaml.NewDecoder(f).Decode(&s)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse service.yaml in dependency %s", name)
+		}
+
+		foundDeps = append(s.Dependencies.Required, s.Dependencies.Optional...)
+	} else {
+		log.Info().Msgf("Using baked-in dependency list")
+		foundDeps = virtualDeps[name]
+	}
+
+	for _, d := range foundDeps {
+		err := grabDependencies(ctx, deps, d, auth)
+		if err != nil {
+			return err
+		}
+	}
+
+	deps[name] = true
+
+	return nil
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
+
+	log.Info().Msg("Building dependency tree")
+
+	deps, err := BuildDependenciesList(ctx)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to build dependency tree")
+	}
+
+	log.Info().Strs("deps", deps).Msg("Provisioning devenv")
+
+	// TODO: outreach specific code
+	target := "base"
+	for _, d := range deps {
+		if d == "flagship" {
+			target = "flagship"
+			break
+		}
+	}
+
+	//nolint:errcheck // Why: Best effort remove existing cluster
+	exec.CommandContext(ctx, "devenv", "--skip-app", "destroy").Run()
+
+	cmd := exec.CommandContext(ctx, "devenv", "--skip-update", "provision", "--snapshot-target", target)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	err = cmd.Run()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to provision devenv")
+	}
+
+	for _, d := range deps {
+		// Skip dep with same name as our target, e.g. flagship
+		if d == target {
+			continue
+		}
+
+		log.Info().Msgf("Deploying dependency '%s'", d)
+		cmd := exec.CommandContext(ctx, "devenv", "--skip-update", "deploy-app", d)
+		cmd.Stderr = os.Stderr
+		cmd.Stdout = os.Stdout
+		cmd.Stdin = os.Stdin
+		err = cmd.Run()
+		if err != nil {
+			log.Fatal().Err(err).Msgf("Failed to deploy dependency '%s'", d)
+		}
+	}
+
+	log.Info().Msg("Deploying current application into cluster")
+	cmd = exec.CommandContext(ctx, "devenv", "--skip-update", "deploy-app", ".")
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	err = cmd.Run()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to deploy current application into devenv")
+	}
+
+	log.Info().Msg("Waiting for application to be ready")
+	time.Sleep(30 * time.Second) // TODO: Eventually actually wait
+
+	log.Info().Msg("Starting devenv tunnel")
+	cmd = exec.CommandContext(ctx, "devenv", "--skip-app", "tunnel")
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to start devenv tunnel")
+	}
+	time.Sleep(30 * time.Second) // TODO: Localizer should expose an event when "ready"
```

Comment:
maybe we should cut a ticket for this just to remember. possibly same with the TODO directly above this relating to waiting for the devenv to finish deploying an application